Coupled-Cluster/RHF/helper_cchbar.py

```diff
@@ -0,0 +1,254 @@
+"""
+A simple python script to build pieces of the similarity transformed hamiltomian,
+Hbar = e^(-T)He^(T) = H + [H,T] + 1/2![[H,T],T] + 1/3![[[H,T],T],T] + 1/4![[[[H,T],T],T],T]
+which can be used quite conveniently to solve lambda equations, calculate  excitation energies 
+(EOM-CC sigma equations) , CC response properties etc..
+
+References:  
+1. J. Gauss and J.F. Stanton, J. Chem. Phys., volume 103, pp. 3561-3577 (1995). 
+"""
+
+__authors__ = "Ashutosh Kumar"
+__credits__ = ["T. D. Crawford", "Daniel G. A. Smith", "Lori A. Burns", "Ashutosh Kumar"]
+
+__copyright__ = "(c) 2014-2017, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2017-05-17"
+
+import time
+import numpy as np
+import psi4
+from utils import ndot
+
+class HelperCCHbar(object):
+
+    def __init__(self, ccsd, memory=2):
+
+        # Start of the cchbar class
+        time_init = time.time()
+
+        self.MO = ccsd.MO
+        self.ndocc = ccsd.ndocc
+        self.nmo = ccsd.nmo
+        self.nocc = ccsd.ndocc
+        self.nvirt = ccsd.nmo - ccsd.nocc
+
+        self.slice_o = slice(0, self.nocc)
+        self.slice_v = slice(self.nocc, self.nmo)
+        self.slice_a = slice(0, self.nmo)
+        self.slice_dict = {'o' : self.slice_o, 'v' : self.slice_v,
+                           'a' : self.slice_a}
+
+
```

Comment:
Can you run all of these files through `yapf`?

Reply:
Do you mean using the default yapf settings. Because doing a `yapf -i helper_cchbar.py`
breaks the ndot calls into two lines. Is this acceptable ? Ex. 
``` 
self.Hoooo += ndot('ijef,mnef->mnij', self.build_tau(),
                           self.get_MO('oovv'))
```

Reply:
if any particular lines get too mangled by yapf, add `  # yapf: disable` at their ends. a mild sprinkling of these (I think I average one per file) can help readability.

Reply:
Done! if you are fine with this format.